lib/task-service.js

```diff
@@ -4,21 +4,54 @@
 
 var di = require('di');
 
-module.exports = factory;
-di.annotate(factory, new di.Provide('Services.Task'));
-di.annotate(factory, new di.Inject(
+module.exports = taskServiceFactory;
+di.annotate(taskServiceFactory, new di.Provide('Services.Task'));
+di.annotate(taskServiceFactory, new di.Inject(
+    'Logger',
+    'Promise',
+    'ChildProcess',
     'TaskOption.Validator'
 ));
 
-function factory(validator) {
+function taskServiceFactory(
+    Logger,
+    Promise,
+    ChildProcess,
+    validator
+) {
+    var logger = Logger.initialize(taskServiceFactory);
+    var requiredCmdTools = ['ipmitool'];
+
     function TaskService() {}
 
     TaskService.prototype.start = function() {
-        return validator.register();
+        var self = this;
+        return Promise.resolve()
+        .then(function() {
+            return self.checkToolsExist();
+        })
+        .then(function() {
```

Comment:
Should this be a tap?  Do you want to return the value that came out of checkToolsExist or the value coming out of the validator register?